### PR TITLE
[mqtt.espmilighthub] Fix things do not come ONLINE if MQTT Client Status Topic not supplied.

### PR DESCRIPTION
**Pull request by Skinah**
Added the extra needed steps to the readme file and also helpful thing status messages to guide a user on what is now needed after PR #13800

Without doing extra setup steps, the last PR has caused all the things to stay on initializing status as the connected MQTT message that is needed never gets received.

Signed-off-by: Matthew Skinner <matt@pcmus.com>

### CODE REVIEW

**Review by ccutrer — APPROVED** [review]
🤦  I don't even remember configuring that on my MilightHub (and that it wasn't in this readme already)

File: bundles/org.openhab.binding.mqtt.espmilighthub/src/main/java/org/openhab/binding/mqtt/espmilighthub/internal/handler/EspMilightHubHandler.java [hunk 1]
```diff
@@ -537,6 +537,8 @@ public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
                 return;
             }
             this.connection = connection;
+            updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NOT_YET_READY,
+                    "Waiting for 'milight/status: connected' MQTT message to be recieved. Check hub has 'MQTT Client Status Topic' configured.");
```
Comment (lolodomo) [inline bundles/org.openhab.binding.mqtt.espmilighthub/src/main/java/org/openhab/binding/mqtt/espmilighthub/internal/handler/EspMilightHubHandler.java:541]: Typo: received

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.mqtt.espmilighthub/src/main/java/org/openhab/binding/mqtt/espmilighthub/internal/handler/EspMilightHubHandler.java (hunk 1 above)
Comment (jlaur) [reply bundles/org.openhab.binding.mqtt.espmilighthub/src/main/java/org/openhab/binding/mqtt/espmilighthub/internal/handler/EspMilightHubHandler.java:541]: I believe `ThingStatusDetail.NOT_YET_READY` was introduced by #3330 and used in combination with `ThingStatus.UNINITIALIZED` by the framework:

https://github.com/openhab/openhab-core/blob/cdab53b814c08380bba52a3f5f31543117be52cb/bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingManagerImpl.java#L926-L931

So I'm not sure it's intended the be used by bindings explicitly and in this combination with `ThingStatus.UNKNOWN`. Maybe the documentation needs to be updated: https://next.openhab.org/docs/concepts/things.html#status-details

@J-N-K - perhaps you can comment?

Comment (lolodomo) [reply bundles/org.openhab.binding.mqtt.espmilighthub/src/main/java/org/openhab/binding/mqtt/espmilighthub/internal/handler/EspMilightHubHandler.java:541]: I think you're right @jlaur

Comment (lolodomo) [reply bundles/org.openhab.binding.mqtt.espmilighthub/src/main/java/org/openhab/binding/mqtt/espmilighthub/internal/handler/EspMilightHubHandler.java:541]: I just grep our binding database and it is clean, NOT_YET_READY is not used.

**Comment by lolodomo** [comment]
@Skinah : did you see the review comments ?

**Comment by Skinah** [comment]
@lolodomo I made some changes that should resolve all issues. Been very busy and travelling recently.

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Comment by lolodomo** [comment]
Build is failing due to org.openhab.binding.mqtt.ruuvigateway.tests not found. That is normal, we disabled them as a change is required to this project.
@openhab/add-ons-maintainers : I am not sure if we can just ignore this build error ?

**Review by J-N-K — APPROVED** [review]
LGTM

**Review by J-N-K — APPROVED** [review]
LGTM

**Pull request merged**